x-pack/plugins/maps/public/components/map/mb/view.js

```diff
@@ -6,154 +6,252 @@
 
 import _ from 'lodash';
 import React from 'react';
+import ReactDOM from 'react-dom';
 import { ResizeChecker } from 'ui/resize_checker';
 import { syncLayerOrder, removeOrphanedSourcesAndLayers, createMbMapInstance } from './utils';
 import { DECIMAL_DEGREES_PRECISION, ZOOM_PRECISION } from '../../../../common/constants';
 import mapboxgl from 'mapbox-gl';
+import { FeatureTooltip } from '../feature_tooltip';
 
 export class MBMapContainer extends React.Component {
 
   constructor() {
     super();
     this._mbMap = null;
+    this._tooltipContainer = document.createElement('div');
     this._listeners = new Map(); // key is mbLayerId, value eventHandlers map
+    this._mbPopup = new mapboxgl.Popup({
+      closeButton: false,
+      closeOnClick: false,
+    });
   }
 
   _debouncedSync = _.debounce(() => {
     if (this._isMounted) {
       this._syncMbMapWithLayerList();
       this._syncMbMapWithInspector();
+      this._syncTooltipState();//consider not debouncing
```

Comment:
Lets move this out of debounce to be more responsive. The debounce delay is pretty noticeable for tooltips

Reply:
Plus, the action is already debounced by mousemove callback